### ISSUE TITLE
Move Sequencescape API v1 calls over to API v2

### ISSUE THREAD

**Issue by sdjmchattie** [sanger/gatekeeper#410]
**User story**

Sequencescape API v1 will be shut down in the near future and has been deprecated for the past 5 years.  We should switch usage of the API over to v2 by installing and making use of the JSONAPI:Resources gem.

**Who are the primary contacts for this story**

Stuart McHattie

**Who is the nominated tester for UAT**

PSD

**Acceptance criteria**

To be considered successful the solution must allow:

- [ ] No more API v1 calls are made to Sequencescape
- [ ] All traces of `sequencescape-client-api` and its config have been removed.

**Dependencies**
Any new endpoints required in Sequencescape will have to be created as they are discovered to be missing.  A new story should be created under that repository if there are indeed endpoints to create.

**Additional context**

The calls made during the last two week period are as below, with counts for each endpoint:

- GET    - /api/1/: 33
- GET    - /api/1/tag_layout_templates: 4
- GET    - /api/1/tag_layout_templates/2: 4
- GET    - /api/1/{uuid}: 45
- GET    - /api/1/{uuid}/qcables: 12

- POST   - /api/1/qcable_creators: 4
- POST   - /api/1/{uuid}/first: 17
- POST   - /api/1/{uuid}/lots: 4

**Comment by sdjmchattie** [comment]
Reviewing the code, there are v1 endpoints being called in a lot of places, which is not reflected in the endpoints actually being hit on Sequencescape as shown above.  The working assumption is that Gate Keeper is not used as much any more and only some parts of it are in active use.  To track, though, the obvious uses of v1 endpoints are for:

- Creating state changes.
- Finding assets by barcode.
- Creating QC decisions.
- Finding lot types by various criteria.
- Finding plates by ID.
- Finding lots by ID.
- Creating QC Creators.
- Finding robots by barcode.
- Creating stamps.
- Find QCAbles by barcode.
- Finding robots by UUID.
- Finding robots by barcode.
- Finding order templates and creating orders from them.
- Creating submissions.
- Finding tubes by ID.
- Finding users by swipecard.
- Finding barcode printers.
- Finding lots by batch ID.
- Finding transfer templates.
- Getting all tag layout templates.
- Getting all tag2 layout templates.
- Creating bulk transfers.
- Finding tag/tag2 layout templates and then creating a tag layout from it.
- Creating plate conversions.
- Creating plates.
- Creating specific tubes.
- Creating tubes from tubes.
- Getting all plate templates.
- Finding plate purposes.
- Finding tube purposes.
- Fetching studies and projects.

These may not be exhaustive, but I think they are.  The stakeholders will be consulted on this as this would be a lot of work to switch over to V2 and it looks like most of this isn't actually used.

**Comment by stevieing** [comment]
We will be starting communications soon to review the usage of Gatekeeper with a view to deprecating.

**Comment by KatyTaylor** [comment]
We prioritised this today because:
1. It helps towards getting rid of SS API v1, and this is currently causing extra complexity for this dependency update - https://github.com/sanger/sequencescape/pull/4970
2. We won't start 'declining' Gatekeeper for a little while, because we're focusing on declining Heron services first.
3.  Declining Gatekeeper might take a while because we have to replace the functionality, on top of the comms.
4. Doing this story may help in declining Gatekeeper, by adding to our team understanding of the app.

**Comment by KatyTaylor** [comment]
Putting this back in the backlog, as the 'epic' story, which is not yet complete. The sub-issues should be done first, and then this issue can be used to consider if anything else needs to be done (e.g. remove any remaining API v1 traces from Gatekeeper).

**Comment by StephenHulme** [comment]
Most recent tally of API v1 calls post Lots changes (2026-05-14):
```md
| Normalised path                     | Client          | Count |
| ----------------------------------- | --------------- | ----- |
| /api/<id>/                          | gate-prod-jammy | 48    |
| /api/<id>/<uuid>                    | gate-prod-jammy | 16    |
| /api/<id>/<uuid>/first              | gate-prod-jammy | 16    |
| /api/<id>/tag_layout_templates      | gate-prod-jammy | 4     |
| /api/<id>/tag_layout_templates/<id> | gate-prod-jammy | 4     |
```
